awx/main/models/unified_jobs.py

```diff
@@ -1484,3 +1489,17 @@ def is_isolated(self):
     @property
     def is_containerized(self):
         return False
+
+    def log_lifecycle(self, state, blocked_by=None):
+        extra={'type': self._meta.model_name,
+               'task_id': self.id,
+               'state': state}
+        if self.unified_job_template:
+            extra["template_name"] = self.unified_job_template.name
+        if state == "blocked" and blocked_by:
+            blocked_by_msg = f"{blocked_by._meta.model_name}-{blocked_by.id}"
```

Comment:
We've got a property that's pretty similar to this that we use in other logging contexts:

`UnifiedJob.log_format`

It also includes the current status of the job.  If that's too verbose, that's fine, but it might be an opportunity for reuse here.

Reply:
```In [1]: Job.objects.first().log_format
Out[1]: 'job 1 (successful)'```

Reply:
```In [4]: [j.log_format for j in UnifiedJob.objects.all()]
Out[4]:
['job 1 (successful)',
 'project_update 2 (successful)',
 'ad_hoc_command 3 (successful)',
 'ad_hoc_command 4 (successful)',
 'ad_hoc_command 5 (successful)',
 'job 6 (successful)',
 'project_update 7 (successful)']```

Reply:
You know @fosterseth ignore me, I like the way yours looks better.